application/electron/src/services/files/service.file.recent.ts

```diff
@@ -0,0 +1,245 @@
+                        if (err) {
+                            return resolveFile();
+                        }
+                        file.size = stats.size;
+                        checked.push(file);
+                        resolveFile();
+                    });
+                });
+            })).then(() => {
+                if (checked.length === stored.recentFiles.length) {
+                    return resolve(checked);
+                }
+                ServiceStorage.get().set({
+                    recentFiles: checked,
+                });
+                ServiceElectron.updateMenu();
+                resolve(checked);
+            });
+        });
+    }
+
+    public clear() {
+        ServiceStorage.get().set({
+            recentFiles: [],
+        });
+        ServiceElectron.updateMenu();
+    }
+
+    private _ipc_FilesRecentRequest(request: IPCMessages.TMessage, response: (instance: IPCMessages.TMessage) => any) {
+        this.get().then((files: IStorageScheme.IRecentFile[]) => {
+            files.sort((a: IStorageScheme.IRecentFile, b: IStorageScheme.IRecentFile) => {
+                return a.timestamp < b.timestamp ? 1 : -1;
+            });
+            response(new IPCMessages.FilesRecentResponse({
+                files: files,
+            }));
+        });
+    }
+
+    private _validate(): Promise<IStorageScheme.IRecentFilterFile[]> {
+        return new Promise((resolve) => {
+            const stored: IStorageScheme.IStorage = ServiceStorage.get().get();
+            const files: IStorageScheme.IRecentFilterFile[] = [];
+            Promise.all(stored.recentFiltersFiles.map((file: IStorageScheme.IRecentFilterFile) => {
+                return new Promise((resolveFile) => {
+                    fs.access(file.file, fs.constants.F_OK, (err) => {
+                        if (err) {
+                            return resolveFile();
+                        }
+                        files.push(file);
+                        resolveFile();
+                    });
+                });
+            })).then(() => {
+                if (files.length === stored.recentFiltersFiles.length) {
+                    return resolve(files);
+                }
+                ServiceStorage.get().set({
+                    recentFiltersFiles: files,
+                });
+                ServiceElectron.updateMenu();
+                resolve(files);
+            });
+        });
+    }
+
+    private _ipc_onFiltersRecentRequested(message: IPCMessages.TMessage, response: (message: IPCMessages.TMessage) => Promise<void>) {
+        this._validate().then((files: IStorageScheme.IRecentFilterFile[]) => {
+            files.sort((a: IStorageScheme.IRecentFilterFile, b: IStorageScheme.IRecentFilterFile) => {
+                return a.timestamp < b.timestamp ? 1 : -1;
+            });
+            response(new IPCMessages.FiltersFilesRecentResponse({
+                files: files,
+            }));
+        });
```

Comment:
if `_validate` never reject - it's ok do not have here `catch`, but validate doesn't have `catch` section and it could be possible to be rejected